### PR TITLE
doc: add thread and zigbee multiprotocol guide

### PR DESCRIPTION
**Pull request by greg-fer**
Added document that describes how to enable multiprotocol support.
KRKNWK-6820

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/ug_multiprotocol_support.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
+.. _ug_multiprotocol_support:
+
+Multiprotocol support for Thread and Zigbee
+###########################################
+
+The :ref:`ug_thread` and :ref:`ug_zigbee` protocols support running another protocol in parallel.
+
+This is achieved with the :ref:`nrfxlib:softdevice_controller`, an implementation of the :ref:`ug_ble_controller`.
+The SoftDevice Controller in turn relies on the functionalities of ref:`Multi-Protocol Service Layer (MPSL) <nrfxlib:mpsl>` library, which provides services for multiprotocol applications and allows the 802.15.4 radio driver to negotiate for transmission timeslots with the controller.
```
Comment (hubertmis) [inline doc/nrf/ug_multiprotocol_support.rst:9]: Actually, it is achieved by the implementation of MPSL. Any radio protocol using MPSL to schedule radio access can be used in parallel with thread and zigbee. Sd controller is just one example of such protocol. I mean, sd controller is not needed for parallel radio operations. Only MPSL is needed. SD controller is just an example of other radio protocol that can run in parallel with thread or zigbee

**Review by hubertmis — COMMENTED** [review]
One sentence in the doc is misleading. I would rephrase it.

File: doc/nrf/ug_multiprotocol_support.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+For more information about how the SoftDevice Controller and MPSL work together, read the :ref:`SoftDevice Controller's documentation <softdevice_controller_readme>`.
+
+The multiprotocol solution available for Thread and Zigbee is dynamic.
+This means that it allows for running several radio protocols simultaneously without the time-expensive uninitialization and initialization in-between the switching.
+Switching between protocols requires only a reinitialization of the radio peripheral, since protocols may operate on different frequencies, modulations, etc.
+
+Enabling multiprotocol support
+******************************
+
+To enable the multiprotocol support for either Thread or Zigbee, set the following Kconfig option:
```
Comment (lmaciejonczyk) [inline doc/nrf/ug_multiprotocol_support.rst:19]: @hubertmis Is it still valid statement that the new 802.15.4 radio driver will by default activate MPSL and enable multiprotocol support in that way?

File: doc/nrf/ug_multiprotocol_support.rst (hunk 1 above)
Comment (lmaciejonczyk) [reply doc/nrf/ug_multiprotocol_support.rst:9]: I guess that SD controller is the only one which uses MPSL in NCS. We can not run for instance Zephyr BLE controller together with thread or zigbee. From what I understand in theory someone can provide it's own protocol stack which uses MPSL and based on that can run thread/zigbee concurrently.

File: doc/nrf/ug_multiprotocol_support.rst [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+Given the nature of the dynamic multiprotocol solution, take into account the following considerations when developing your Thread or Zigbee application:
+
+- Any kind of application or Thread/Zigbee stack logic can be interrupted by the |BLE| activity.
+  This is the primary difference between a Thread/Zigbee-only application and a multiprotocol one.
+  As a result, it is acceptable that the node can sometimes miss a frame and not respond with a MAC-level acknowledgment.
+- |BLE| scanning requires a lot of time to process |BLE| traffic and thus blocks 802.15.4 traffic.
+  When the scanning is used, it is recommended to use the Sleepy End Device role for Thread or Zigbee applications.
+- Be aware that flash operations require more time to complete.
+  A flash operation may be performed only in a free timeslot, so it has to wait until the current timeslot finishes (it may be a 802.15.4 timeslot or a |BLE| one).
+  Moreover, the Thread or Zigbee stacks cannot operate while the flash operation is being performed.
```
Comment (lmaciejonczyk) [inline doc/nrf/ug_multiprotocol_support.rst:40]: Just as a note, at the moment, SoftDevice Contoller provides implementation of the flash driver which synchronize with it. This feature is activated if SoftDevice Contoller is enabled. The work to provide common flash radio scheduler interface is in progress.

File: doc/nrf/ug_multiprotocol_support.rst (hunk 1 above)
Comment (greg-fer) [reply doc/nrf/ug_multiprotocol_support.rst:9]: Rephrased.

Comment (hubertmis) [reply doc/nrf/ug_multiprotocol_support.rst]: Yes, the only protocols provided by nordic in ncs that support f dynamic multiprotocol are 802.15.4 driver and sd controller. However, a customer could implement a proprietary radio protocol using Timeslot API from MPSL and use dynamic multiprotocol with 802.15.4, without BLE controller.
Because of that I suggest to remove text indicating that dynamic multiprotocol is implemented in SD Ctlr.

File: doc/nrf/ug_multiprotocol_support.rst (hunk 2 above)
Comment (hubertmis) [reply doc/nrf/ug_multiprotocol_support.rst:19]: To enable multiprotocol support in the 802.15.4 driver MPSL must be enabled. Enabling SD controller only enables another protocol working in parallel. It is not necessary for multiprotocol applications. It's just an example.

**Review by LuDuda — APPROVED** [review]

**Review by maciekfabia — APPROVED** [review]

**Review by lmaciejonczyk — APPROVED** [review]
